rinja_derive/src/generator.rs

```diff
@@ -88,6 +89,23 @@ impl<'a> Generator<'a> {
         Ok(buf.buf)
     }
 
+    fn generate_include_bytes<S: AsRef<str>>(&mut self, buf: &mut Buffer, path: S) {
+        static CACHE: OnceLock<OnceMap<String, &'static ()>> = OnceLock::new();
```

Comment:
Thinking about it, I don't think using a cache this why is sane:

* Given the file `layout.html` is used in compiling units `a.rs` and `b.rs`, then the include might wind up only in one of the generated objects, say, `a.rs`.
* Afterwards, we remove `a.rs` and any mentions to it.
* Then we alter `layout.html`: `b.rs` does not mention the dependency, so the change won't cause a recompilation.

Reply:
If we remove `a.rs`, the crate will be recompiled and `layout.html` will be used in `b.rs`. Or did I miss something?

Reply:
You cannot assume that the whole crate is recompiled when incremental compilation is enabled, can you?

Reply:
Hum... It's a good question. But since you remove `a.rs`, it also means you remove `mod a` somewhere else, so in this case, I wonder if the whole crate isn't simply rebuilt...

Reply:
But what if the add more indirection? (Assume the usual folder structure):

```rust
mod x1 {
    mod x2 {
        mod a;
    }
    mod x3 {
        mob b;
    }
}
```

Then removing `mod a` while cause a recompilation of `mod x2`, but not `x1`. If the need to rebuild would bubble up, then there could be no incremental compilation.

Reply:
The fact that we're debating over this is already a pretty good sign that we should not do it. Closing then! :)

Reply:
Maybe once upon a time <https://github.com/rust-lang/rust/issues/99515> gets implemented. :) If I understand the proposal correctly, then this could be a game changer, and we could even use spans inside the template file to show error messages, etc.